File: examples/src/bin/delete_user.rs
use objectscale_client::client::ManagementClient;

fn main() {
    let endpoint = "https://10.225.108.189:443";
    let username = "root";
    let password = "Password123@";
    let insecure = true;

    let user_name = "test";
    let namespace = "osai0a9250592a131336";

    let mut client: ManagementClient =
        ManagementClient::new(endpoint, username, password, insecure).expect("management client");
    client
        .delete_user(user_name, namespace)
        .expect("delete user");
    println!("Deleted user: {}", user_name);
}

File: golang/examples/delete_user.go
package main

import (
	"log"

	objectscale "github.com/vangork/objectscale-client/golang/pkg"
)

func main() {
	endpoint := "https://10.225.108.217:4443"
	username := "root"
	password := "Password123!"
	insecure := true

	client, err := objectscale.NewManagementClient(endpoint, username, password, insecure)
	if err != nil {
		log.Panicln("Fail to create objectscale client:", err.Error())
	}
	defer client.Close()

	name := "luis_user"
	namespace := "ns1"
	err = client.DeleteUser(name, namespace)
	if err != nil {
		log.Println(err)
	} else {
		log.Printf("Deleted user: %s\n", name)
	}
}
